indico/modules/rb/controllers/backend/bookings.py

```diff
@@ -422,6 +423,31 @@ def _process(self):
         return jsonify(occurrence=reservation_occurrences_schema.dump(self.occurrence, many=False))
 
 
+class RHBookingOccurrenceLinkActions(RHBookingBase):
+    """Links a Reservation occurrence to an event."""
+
+    def _process_args(self):
+        RHBookingBase._process_args(self)
+        date = dateutil.parser.parse(request.view_args['date'], yearfirst=True).date()
+        self.occurrence = self.booking.occurrences.filter(ReservationOccurrence.date == date).one()
+        self.event = Event.query.filter_by(id=request.view_args['event_id'], is_deleted=False).first_or_404()
```

Comment:
We have a shortcut for this:

```suggestion
        self.event = Event.get_or_404(request.view_args['event_id'], is_deleted=False)
```

Reply:
Fixed.